src/libecalc/presentation/yaml/yaml_types/components/legacy/energy_usage_model/yaml_energy_usage_model_consumer_system.py

```diff
@@ -65,6 +65,36 @@ class YamlCompressorSystemOperationalSetting(YamlBase):
         description="Set suction pressure equal for all consumers in a consumer system operational setting. \n\n$ECALC_DOCS_KEYWORDS_URL/SUCTION_PRESSURE",
     )
 
+    @model_validator(mode="after")
+    def ensure_increasing_cross_over(self):
+        if self.crossover is None:
+            return self
+        for compressor_train_index, crossover_to in enumerate(self.crossover):
+            crossover_to_compressor_train_index = (
+                crossover_to - 1
+            )  # compressor_train_index is 0-indexed, crossover_to is 1-indexed
+            no_crossover = crossover_to == 0
+            if crossover_to_compressor_train_index > compressor_train_index or no_crossover:
+                pass  # passing excess rate to a comp. train not yet evaluated or no crossover defined for this comp. train
+            else:
+                raise ValueError(
+                    f"Crossover: {self.crossover}\n"
+                    "The compressor trains in the compressor system are not defined in the correct order, according to "
+                    "the way the crossovers are set up. eCalc can now try to pass excess rates to compressor trains in "
+                    "the system that has already been evaluated. \n\n"
```

Comment:
Is this correct? Or should it be not evaluated?

Reply:
The way it works now if you have crossover: [0, 1, 0], you first evaluate the first consumer (here it has no crossover). The next consumers can pass excess rate to the first one, potentially exceeding the maximum rate there, and the operational setting will be chosen even if it is invalid.

Reply:
The sentence `eCalc can now try to pass excess rates to compressor trains the system that has already been evaluated` confused me. Is the intention to inform the user why the current crossover setting isn't allowed? So `If we were to allow this crossover setup, eCalc™ could try to pass excess rates to compressor trains in the system that has already been evaluated`?

Reply:
Poorly phrased, I guess... Yes, that was happening in one model. The first train is within capacity, the second one outside capacity, so rate was passed to the first one (which then with the additional rate was outside capacity).